generic-gcc-avr.cmake

```diff
@@ -166,26 +166,30 @@ function(add_avr_executable EXECUTABLE_NAME)
       message(FATAL_ERROR "No source files given for ${EXECUTABLE_NAME}.")
    endif(NOT ARGN)
 
+   set(target_name ${EXECUTABLE_NAME}${MCU_TYPE_FOR_FILENAME})
+   set(bin_dir "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE}")
```

Comment:
This is changing the output directory, i.e. it changes behaviour, doesn't it?

Reply:
Yes, it changes build directory, respectively define relation between build directory and output file. In previous case it as relative to executed `cmake` command.

Not saying you have to accept it, but in general, output files shall be defined as inside "build" directory.

Reply:
Normally the user chooses the build directory depending on the needs, like
```
cmake . -Bbuild-firmware
```
where the build is done in `build-firmware`. IMHO the CMake code should respect the choice of the user and not change it in an arbitrary way.